src/classes/emitter.ts

```diff
@@ -1,49 +1,110 @@
 // deno-lint-ignore-file no-explicit-any
+export type EventName = string | symbol;
 export type Listener = (this: Emitter, ...args: any[]) => void;
```

Comment:
I would propose to change it to
```ts
export type Listener<T = any[]> = (...args: T) => void
```

Why do we have `this` here. The listener has not to be bound to the emitter

Reply:
See also https://github.com/mathe42/surrealdb.deno/blob/82dee508405f5662beecaa4624b0dff9bc866afe/src/classes/emitter.ts how I did it.

Reply:
Because it was part of the original code and it's how node's `EventEmitter` handles it

```ts
import { EventEmitter } from 'node:events'

const ee = new EventEmitter()

ee.on('foo', function() {
  console.log(this) // EventEmitter {}
})

ee.emit('foo')
```